CMake/ResolveDependency.cmake

```diff
@@ -129,3 +129,13 @@ function(list_subdirs var dir)
       ${dirs}
       PARENT_SCOPE)
 endfunction()
+
+# Set custom source url with a optional sha256 checksum.
+macro(resolve_dependency_url dependency_name)
+  set_with_default(VELOX_${dependency_name}_SOURCE_URL VELOX_${dependency_name}_URL
+                   ${VELOX_${dependency_name}_SOURCE_URL})
+  if(DEFINED ENV{VELOX_${dependency_name}_URL})
+    set_with_default(VELOX_${dependency_name}_BUILD_SHA256_CHECKSUM VELOX_${dependency_name}_SHA256
+                     "")
```

Comment:
Nice, thanks for your improvements! 
This handles the case of no explicit sha envvar well but would require the user to prepend `SHA256=` when explicitly setting it. Which could be fine but we could also prepend that if the envvar is defined.

It would be great if you could add a section to the README on how to specify a custom url/file path for an offline build in `CMake/resolve_dependency_modules/README.md`. As I neglected that at the time :see_no_evil: :grin:

Reply:
Thanks! I will update README later.